### ISSUE TITLE
.query doesn't seem to work

### ISSUE THREAD

**Issue by caiges** [balderdashy/sails-mysql#245]
In a simplified case:

`User.query('SELECT id FROM users WHERE id = $1', [1], function(...)`

I get:

`Error: ER_BAD_FIELD_ERROR: Unknown column '$1' in 'where clause'`

According to the [docs](https://github.com/balderdashy/waterline-docs/blob/master/queries/query-methods.md#query-query-data-callback-), this should work. Thoughts?

**Comment by sailsbot** [comment]
Thanks for posting, @caiges.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by Wizzercn** [comment]
I also encountered this problem, and find a solution.please..

**Comment by devinivy** [comment]
Does it work if you replace `$1` with `?`?

**Comment by Wizzercn** [comment]
Sys_unit.query('select * from sys_unit where parentId = $1 order by location asc,path asc',[req.params.id],function (err, objs) {
      
    });

err:::{"code":"ER_BAD_FIELD_ERROR","errno":1054,"sqlState":"42S22","index":0}

:(

**Comment by Wizzercn** [comment]
with ? is ok....  thank u..

**Comment by devinivy** [comment]
Yes, I'll make an issue with waterline-docs.

**Comment by Wizzercn** [comment]
parameters,not run:

Sys_unit.query('select * from sys_unit where parentId = ? or id=? order by location asc,path asc',[1,2],function (err, objs) {

});

how to use,please write in the document...

**Comment by Wizzercn** [comment]
Is it safe to pass parameters like this?
You can filter out the character of the unsafe..

ps:I'm sorry, but I'm on the translation software......:)

**Comment by devinivy** [comment]
You probably just need to add a space before the `?`.  It uses the syntax described here: https://github.com/felixge/node-mysql/#escaping-query-values

**Comment by devinivy** [comment]
It is safe!  Internally the mysql driver escapes the values you pass in the second argument.
